client/orm/db.go

```diff
@@ -819,63 +819,121 @@ func (d *dbBase) UpdateBatch(ctx context.Context, q dbQuerier, qs *querySet, mi
 
 	join := tables.getJoinSQL()
 
-	var query, T string
+	sets := d.getSetSQL(columns, values)
+
+	query := d.UpdateBatchSQL(mi, sets, specifyIndexes, join, where)
+
+	res, err := q.ExecContext(ctx, query, values...)
+	if err == nil {
+		return res.RowsAffected()
+	}
+	return 0, err
+}
+
+func (d *dbBase) UpdateBatchSQL(mi *models.ModelInfo, sets, specifyIndexes, join, where string) string {
+	Q := d.ins.TableQuote()
+
+	buf := buffers.Get()
+	defer buffers.Put(buf)
+
+	_, _ = buf.WriteString("UPDATE ")
+	_, _ = buf.WriteString(Q)
+	_, _ = buf.WriteString(mi.Table)
+	_, _ = buf.WriteString(Q)
+
+	if d.ins.SupportUpdateJoin() {
+		_, _ = buf.WriteString(" T0 ")
+		_, _ = buf.WriteString(specifyIndexes)
+		_, _ = buf.WriteString(join)
+		_, _ = buf.WriteString(sets)
+		_, _ = buf.WriteString(" ")
+		_, _ = buf.WriteString(where)
+	} else {
+		_, _ = buf.WriteString(" ")
+		_, _ = buf.WriteString(sets)
+		_, _ = buf.WriteString(" WHERE ")
+		_, _ = buf.WriteString(Q)
+		_, _ = buf.WriteString(mi.Fields.Pk.Column)
+		_, _ = buf.WriteString(Q)
+		_, _ = buf.WriteString(" IN ( ")
+		_, _ = buf.WriteString("SELECT T0.")
+		_, _ = buf.WriteString(Q)
+		_, _ = buf.WriteString(mi.Fields.Pk.Column)
+		_, _ = buf.WriteString(Q)
+		_, _ = buf.WriteString(" FROM ")
+		_, _ = buf.WriteString(Q)
+		_, _ = buf.WriteString(mi.Table)
+		_, _ = buf.WriteString(Q)
+		_, _ = buf.WriteString(" T0 ")
+		_, _ = buf.WriteString(specifyIndexes)
+		_, _ = buf.WriteString(join)
+		_, _ = buf.WriteString(where)
+		_, _ = buf.WriteString(" )")
+	}
+
+	query := buf.String()
+
+	d.ins.ReplaceMarks(&query)
+
+	return query
+}
+
+func (d *dbBase) getSetSQL(cols []string, values []interface{}) string {
```

Comment:
refactor this function:
```go
func (d *dbBase) buildSetSQL(bs Buffer, cols []string, values []any) {

}
```
we want to reuse the buffer to reduce the memory allocation.
And you can still keep passing an empty buffer into your unit tests.
After that, rename the T to `owner` and Q to `quote`, T and Q are not good names.

Reply:
if you don't like the name `owner`, you can give it a new name.

Reply:
Okay.